src/application/gitcoin/hooks/use-recipients.ts

```diff
@@ -0,0 +1,73 @@
+import { useMemo } from 'react';
+
+import { useCommandQuery } from 'shared/messaging';
+import { ScrapingResult } from 'shared/scraping';
+import { isHandleNode } from 'host/twitter'; // TODO: abstract this so it's not specific to Twitter, probably we need to accept it via Properties
+
+import { GetApplicationsCommand } from '../commands';
+import { selectTwitterApplications } from '../utils';
+import { Recipient } from '../types';
+
+interface Properties {
+  users: ScrapingResult[];
+  handle?: string;
+  enabled: boolean;
+}
+
+export const useRecipients = ({ users, handle, enabled }: Properties) => {
+  const getApplicationsQuery = useCommandQuery({
+    command: new GetApplicationsCommand({}),
+    select: selectTwitterApplications,
+    placeholderData: (previousData) => {
+      return previousData;
+    },
+    enabled,
+  });
+
+  const recipients: Recipient[] = useMemo(() => {
+    if (!getApplicationsQuery.data) {
+      return [];
+    }
+
+    return users
+      .map((user) => {
+        const application = getApplicationsQuery.data.find((application) => {
+          return (
+            application.project.metadata.projectTwitter?.toLowerCase() ===
+            user.value.toLowerCase()
+          );
+        });
+
+        if (!application) {
+          return;
+        }
+
+        const { value, node, top } = user;
+        const username = value;
```

Comment:
what do you think about changing this to `const { value: username, node, top } = user;` and removing the line below (`const username = value;`)?